SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_MaterialFarmerTools.h

```diff
@@ -83,7 +83,32 @@ class MaterialFarmerOptions : public GroupOption{
     );
 };
 
-void run_material_farmer(SingleSwitchProgramEnvironment& env, BotBaseContext& context, MaterialFarmerOptions& options);
+class MaterialFarmer_Descriptor : public SingleSwitchProgramDescriptor{
```

Comment:
This breaks the pattern of descriptor + program.

The only part that's being shared is the stats. So pull that out into this file and leave the descriptor in the original file for the MaterialFarmer program.

Reply:
I'm not sure on the best way to do this. I previously had the descriptor in the original Material Farmer program. But then I got linker issues. Because MaterialFarmer requires MaterialFarmerTools's functions (e.g. run_material_farmer), but then MaterialFarmerTools requires MaterialFarmer's Stats.

I suppose one solution is to just merge MaterialFarmer and MaterialFarmerTools once again? Put everything in a single file?

Reply:
What are the linker issues? Unresolved symbols? Or Duplicate symbols?

Reply:
I was able to solve the issue by just moving some of the functions (particularly the ones that used the Stats) from MaterialFarmerTools back to MaterialFarmer.

Reply:
No don't do this. Keep the common stuff in MaterialFarmerTools. What errors are you getting?